iidm/iidm-api/src/main/java/com/powsybl/iidm/network/IdBasedBusRef.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.powsybl.commons.PowsyblException;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * There would be three types of id:
+ * 1. id of equipment:
+ * 2. id of a configured bus itself:
+ * 3. id of branch, in this case, side is required
+ * @author Yichen TANG <yichen.tang at rte-france.com>
+ */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class IdBasedBusRef extends AbstractBusRef {
+
+    private final String id;
+    private final Branch.Side side;
+
+    public IdBasedBusRef(String id) {
+        this.id = Objects.requireNonNull(id);
+        this.side = null;
+    }
+
+    @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
+    public IdBasedBusRef(@JsonProperty("id") String id, @JsonProperty("side") Branch.Side side) {
+        this.id = Objects.requireNonNull(id);
+        this.side = side;
+    }
+
+    @Override
+    Optional<Bus> resolveByLevel(Network network, TopologyLevel level) {
+        if (side == null) {
+            final Identifiable<?> identifiable = network.getIdentifiable(id);
+            if (identifiable == null) {
+                return Optional.empty();
+            }
+            if (identifiable instanceof Bus) {
+                Bus bus = (Bus) identifiable;
+                if (Objects.equals(TopologyLevel.BUS_BRANCH, level)) {
+                    return Optional.ofNullable(bus.getConnectedTerminalStream().map(t -> t.getBusView().getBus()).filter(Objects::nonNull).findFirst().orElse(null));
```

Comment:
Note: you can return the result of `findFirst()`. Also we could reuse the `chooseBusByLevel` here since we rely on the terminal.